src/math_parser_diag.cpp

```diff
@@ -1257,6 +1326,7 @@ std::map<std::string_view, dialogue_func_eval> const dialogue_eval_f{
     { "faction_respect", { "g", 1, faction_respect_eval } },
     { "faction_trust", { "g", 1, faction_trust_eval } },
     { "field_strength", { "ung", 1, field_strength_eval } },
+    { "friends_nearby", { "ung", -1, friends_nearby_eval } },
```

Comment:
You're not using any arguments in the function code (the `params` vector):
```suggestion
    { "friends_nearby", { "ung", 0, friends_nearby_eval } },
```

In addition, you've already added the bits for filtering by attitude so why not use them?
```suggestion
    { "npcs_nearby", { "ung", 0, friends_nearby_eval } },
```
Then use an `attitude` kwarg.